### ISSUE TITLE
Folder contents tree navigation broken in Plone 5.1.2

### ISSUE THREAD

**Issue by tomgross** [plone/Products.CMFPlone#2395]
### What I did:

1. Create a Plone 5.1.2-pending Site.
2. Create a nested structure with some folders and pages
3. Navigation within `folder_contents` to a deeper lever.
4. Click `Edit` Button in toolbar.

### What I expect to happen:

Edit navigated page

### What actually happened:

URL points to the folder the navigation started.

### What version of Plone/ Addons I am using:

Plone 5.1.2-pending (minimal)
No Addons.

This works with Plone 5.1.1.

**Comment by tomgross** [comment]
A JavaScript Error is thrown in folder contents:

![grafik](https://user-images.githubusercontent.com/339384/39519353-b647262a-4e06-11e8-9654-45faa466e194.png)

**Comment by gp54321** [comment]
it's because of this [fix](https://github.com/plone/Products.CMFPlone/commit/a3fb0199a187ebde022b059a9605060ca3067e60#diff-c0797358aafd527ae7b3a81b842b51df)  for https://github.com/plone/Products.CMFPlone/issues/2322

Since I am now an old hand at using plone-compile-resources I can confirm that reverting the patch fixes the UI bug and the javascript console error (if it was not obvious enough)

Unfortunately I don't understand the original bug report :-(. What's a fc-breadcrumb ? what's the difference with a mere breadcrumb ? It must have been clear for the people reviewing this patch but not for me.

**Comment by esteele** [comment]
@tomgross Can you confirm this was fixed in the final 5.1.2 release?

**Comment by tomgross** [comment]
Works OK.

**Comment by tomgross** [comment]
Unfortunately I was wrong. The compiled scripts

`static/plone-logged-in-compiled.js`
and
`static/plone-logged-in-compiled.min.js`

of the `Products.CMFPlone-5.1.2.1-py2.7.egg` still contain the wrong JavaScript. To make it work the plone-logged-in bundle needs to be recompiled.

**Comment by tomgross** [comment]
There is another issue in this context, which is probably not a blocker but still annoying: The status message is not reloaded on navigation in folder contents.

Steps to reproduce:

1. Activate plone.app.iterate
2. Make a nested structure with some folders and documents
3. Create a working copy on an upper level
4. Go to the baseline of the working copy -> You will see a banner that this item is locked, which is fine.
5. Navigate within folder contents -> The banner stays visible, which is annoying.

Here I created a working copy of _a-folder_ and navigated one level down to the child _a-page_
<img width="864" alt="screen shot 2018-06-19 at 16 22 36" src="https://user-images.githubusercontent.com/339384/41603732-db4d5b54-73dd-11e8-9fb8-1dc1a15d8b56.png">

**Comment by MrTango** [comment]
At least the recompiling TTW does not help :(

**Comment by MrTango** [comment]
Whats the current status here?
We have a new release of CMFPlone out, but the issue still the same :(

**Comment by gp54321** [comment]
@MrTango : I don't get it. It was my impression that the fix was already committed in [this change](https://github.com/plone/Products.CMFPlone/commit/ce44eca1f35bbebca5f693ad5e120b6e134ab7df#diff-c0797358aafd527ae7b3a81b842b51df)  and indeed if I recompile bundle as it is as shipped with 5.1.2 it is correct; however your patch seems to apply the change again.
Also it does lot of other changes that are unnecessary to fix #2395 so how can it be a matter of recompilation as said in your changelog ???

**Comment by petschki** [comment]
fixed and merged in #2471 ... hopefully released in 5.1.3(.1) ?

**Comment by agitator** [comment]
Can someone confirm that this fix actually worked?
Just tested it on 5.1 coredev and the actions within the toolbar don't get updated.

Even if I put the loggedin bundle in js dev mode, I got the error from toolbar.js

**Comment by sunew** [comment]
this problem has many manifestations - see also https://github.com/plone/mockup/pull/858

I am currently recompiling the bundles to include https://github.com/plone/Products.CMFPlone/issues/2512

**Comment by 2silver** [comment]
@agitator This line [plone.app.content vocabulary.py:254](https://github.com/plone/plone.app.content/blob/master/plone/app/content/browser/vocabulary.py#L254) breaks my folder_contents and leads to the same JS-Error, can you confirm that?

```
    Traceback (innermost last):  
    Module ZPublisher.Publish, line 138, in publish  
    Module ZPublisher.mapply, line 77, in mapply  
    Module ZPublisher.Publish, line 48, in call_object . 
    Module plone.app.content.browser.vocabulary, line 254, in __call__  
    IndexError: tuple index out of range . 
```

**Comment by runyaga** [comment]
This problem is severe.  A user can accidentally perform operations on an item unintentionally.  e.g. they could enter Folder A, navigation through a buncha folders and end on Folder J, add a piece of content and the content ends up in Folder A instead of the intended context of Folder J.

**Comment by thet** [comment]
Should already be fixed in:
https://github.com/plone/Products.CMFPlone/pull/2398
https://github.com/plone/Products.CMFPlone/pull/2399

But for sure in 5.1.5 with:
https://github.com/plone/Products.CMFPlone/pull/2563
https://github.com/plone/plonetheme.barceloneta/pull/165

If it's not, it might be a resource caching problem, where @sunew knows more about it and we're investigating it tomorrow.

**Comment by sunew** [comment]
@agitator @runyaga 
I can confirm this is fixed on latest 5.1.x
